webapp/components/FilterPosts/FilterPosts.vue

```diff
@@ -0,0 +1,282 @@
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+            <ds-flex v-for="category in chunk[6]" :key="category.id" class="categories-list">
+              <ds-flex>
+                <ds-flex-item width="100%">
+                  <ds-button
+                    :icon="category.icon"
+                    :primary="isActive(category.id)"
+                    @click.stop.prevent="toggleCategory(category.id)"
+                  />
+                  <ds-space margin-bottom="small" />
+                </ds-flex-item>
+                <ds-flex>
+                  <ds-flex-item>
+                    <label>{{ category.name }}</label>
+                  </ds-flex-item>
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+            <ds-flex v-for="category in chunk[7]" :key="category.id" class="categories-list">
+              <ds-flex>
+                <ds-flex-item width="100%">
+                  <ds-button
+                    :icon="category.icon"
+                    :primary="isActive(category.id)"
+                    @click.stop.prevent="toggleCategory(category.id)"
+                  />
+                  <ds-space margin-bottom="small" />
+                </ds-flex-item>
+                <ds-flex>
+                  <ds-flex-item>
+                    <label>{{ category.name }}</label>
+                  </ds-flex-item>
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+        </ds-flex>
+      </ds-container>
+    </template>
+  </dropdown>
+</template>
+<script>
+import Dropdown from '~/components/Dropdown'
+import { filterPosts } from '~/graphql/PostQuery.js'
+import { mapMutations } from 'vuex'
+
+export default {
+  components: {
+    Dropdown,
+  },
+  props: {
+    placement: { type: String, default: 'bottom-start' },
+    offset: { type: [String, Number], default: '16' },
+    categories: { type: Array, default: () => [] },
+  },
+  data() {
+    return {
+      pageSize: 12,
+      selectedCategoryIds: [],
+      allCategories: true,
+    }
+  },
+  computed: {
+    chunk() {
+      const groupedCategories = []
+      for (let i = 0; i < this.categories.length; i++) {
+        const last = groupedCategories[groupedCategories.length - 1]
+        if (!last || last.length === 2) {
+          groupedCategories.push([this.categories[i]])
+        } else {
+          last.push(this.categories[i])
+        }
+      }
+      return groupedCategories
+    },
+  },
+  methods: {
+    ...mapMutations({
+      setPosts: 'posts/SET_POSTS',
+    }),
```

Comment:
setting `Posts` in `Vuex` allows us to be able to change the `Posts` in the newsfeed without needing to use a child component.

Reply:
I don't quite get the reason for using Vuex here, could you explain that in more detail?

Reply:
Well, I think I can, yes. The first thing I tried is to add the filter to the `FilterPosts.vue` and then update the apollo cache when the query returned successfully, but I had several issues.

One is that when I added `this.$apollo.query` and tried to update the stores cache it didn't update the `Posts` at all. I was able to get it to work with `this.$apollo.mutate`, even though it is actually a query and doesn't mutate anything, but it only worked if the user double-clicked on the category in the dropdown, which sent two network requests, but updated the `Posts` on the second.

I wanted to make use of the `Post` query with built-in filter that I think you added as part of the `Your filter bubble`, but since `FilterPosts` is a child of `default.vue`, and not a child of `webapp/pages/index.vue`, it's not possible to call `changeFilterBubble(filter)` without using global even listeners, which we've decided to refactor out and not use going forward.

I placed the filter posts dropdown in the `FilterMenu.vue`, and got it working fine, but in a pairing session with @alina-beck, she gave the feedback that it is probably better in the navbar, as filtering is a kind of searching, and it might not be clear what it is in the `FilterMenu`. I tend to agree with her.

Enter Vuex, I am quite familiar with using Redux as a way of managing state and passing props from related but not parent-child components, so I refactored the code to introduce the store where we can update the store after filtering for `Posts`, which in turn updates the `UI`.